### PR TITLE
Small performance enhancements

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
test performance please

File: compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala [hunk 1]
```diff
@@ -44,11 +44,11 @@ class InterceptedMethods extends MiniPhase {
 
   override def phaseName: String = InterceptedMethods.name
 
-  private[this] var primitiveGetClassMethods: Set[Symbol] = _
+  private[this] var scalaValueClasses: scala.collection.Set[Symbol] = _
 
   /** perform context-dependant initialization */
   override def prepareForUnit(tree: Tree)(implicit ctx: Context) = {
-    primitiveGetClassMethods = Set[Symbol]() ++ defn.ScalaValueClasses().map(x => x.requiredMethod(nme.getClass_))
```
Comment (biboudis) [inline compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala]: So, the improvement is that before `prepareForUnit` returned all the symbols for `getClass` methods, by generating those stubsymbols that are needed, and now it doesn't?

File: compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala [hunk 2]
```diff
@@ -105,7 +105,7 @@ class InterceptedMethods extends MiniPhase {
               List(qual, typer.resolveClassTag(tree.pos, qual.tpe.widen))))
           }*/
          */
-      case t if primitiveGetClassMethods.contains(t) =>
+      case t if t.name == nme.getClass_ && scalaValueClasses.contains(t.owner) =>
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala]: I think you could just do defn.ScalaValueClasses.contains, no need for the intermediate variable.

File: compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala (hunk 1 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/InterceptedMethods.scala]: Yes. Mostly the cost was in creating the set.

**Comment by nicolasstucki** [comment]
test performance please

**Comment by nicolasstucki** [comment]
There is some performance regression on some of the exhaustivity benchmarks

**Comment by nicolasstucki** [comment]
Actually the performance regression was on master.

**Pull request merged**